### PR TITLE
Removes fabricSetupFailed.

### PR DESCRIPTION
**Pull request by damencho**
Removes fabricSetupFailed and adds iceConnectionFailure for callstats.

### CODE REVIEW

File: JitsiConference.js [hunk 1]
```diff
@@ -1273,9 +1273,11 @@ function setupListeners(conference) {
                     session, conference.settings);
             });
 
-        conference.room.addListener(XMPPEvents.CONFERENCE_SETUP_FAILED,
-            function () {
-                conference.statistics.sendSetupFailedEvent();
+        conference.room.addListener(XMPPEvents.CONNECTION_ICE_FAILED,
```
Comment (hristoterezov) [inline JitsiConference.js]: Doesn't that mean that we aren't going to log the other cases when we receive XMPPEvents.CONFERENCE_SETUP_FAILED?

Comment (damencho) [inline JitsiConference.js]: Yes, this is intentional. The idea is that we are getting rid of setupFailedEvent that we used to use for callstats and is fired every time on CONFERENCE_SETUP_FAILED event. I've checked and we fire this event(CONFERENCE_SETUP_FAILED) on error in oniceconnectionstatechange, createAnswer, setLocalDescription, setRemoteDescription. And in all cases there are corresponding callstats errors we log except oniceconnectionstatechange. And as callstats.io had introduced new error type called iceConnectionFailure, I've made new event XMPPEvents.CONNECTION_ICE_FAILED and when this happen we log that error and fire CONFERENCE_SETUP_FAILED as we used to before my modifications.

**Comment by damencho** [comment]
Here are my observations:
Currently the only failed fabric event is fabricSetupFailed. It is logged on XMPPEvents.CONFERENCE_SETUP_FAILED.
Where this one is emitted after following failures:

- peerconnection.oniceconnectionstatechange iceConnectionState.failed
- peerconnection.createAnswer error
(on this error there is another event - XMPPEvents.CREATE_ANSWER_FAILED and we report: _reportError(wrtcFuncNames.createAnswer))
- peerconnection.setLocalDescription error
(on this error there is another event - XMPPEvents.SET_LOCAL_DESCRIPTION_FAILED and we report: _reportError(wrtcFuncNames.setLocalDescription))
- peerconnection.setRemoteDescription -> JingleSessionPC.onJingleFatalError
(on this error there is another event - XMPPEvents.SET_REMOTE_DESCRIPTION_FAILED and we report: _reportError(wrtcFuncNames.setRemoteDescription))

**Comment by paweldomas** [comment]
CONFERENCE_SETUP_FAILED is also generated in few places of JingleSessionPC

**Comment by damencho** [comment]
Yes, and as described these all are already covered by callstats.

In JingleSessionPC these are failures of: peerconnection.createAnswer,
peerconnection.setLocalDescription,
peerconnection.setRemoteDescription. Those already fire events which
we listen for and log to callstats. The events though are fired from
TraceablePeerConnection (CREATE_ANSWER_FAILED,
SET_LOCAL_DESCRIPTION_FAILED and SET_REMOTE_DESCRIPTION_FAILED).
The only one missing is for ice failure, which is added in this PR.

**Pull request merged**